libs/common/src/change-stream/delete-one.test.ts

```diff
@@ -0,0 +1,139 @@
+import { when } from 'jest-when';
+import { ObjectId } from 'mongodb';
+import { MongoDbClient } from '../mongo-db-client';
+import { generateNoUserLoggedInAuditDetails } from './generate-audit-details';
+import { MONGO_DB_COLLECTIONS } from '../constants';
+import { generateMockNoUserLoggedInAuditDatabaseRecord } from './test-helpers';
+import { deleteOne } from './delete-one';
+
+const mockSession = {
+  withTransaction: jest.fn((callback: () => void) => {
+    callback();
+  }),
+  endSession: jest.fn(),
+};
+
+const mockClient = {
+  startSession: jest.fn(() => mockSession),
+};
+
+const mockGetCollection = jest.fn();
+
+const mockDeleteOne = jest.fn();
+const mockUsersCollection = {
+  deleteOne: mockDeleteOne,
+};
+
+const mockInsertOne = jest.fn();
+const mockDeletionsCollection = {
+  insertOne: mockInsertOne,
+};
+
+const mockDb = {
+  getClient: jest.fn(() => mockClient),
+  getCollection: mockGetCollection,
+} as unknown as MongoDbClient;
+
+describe('deleteOne', () => {
+  const documentId = new ObjectId();
+  const originalDeletionAuditLogsDeleteAfterSeconds = process.env.DELETION_AUDIT_LOGS_TTL_SECONDS;
+  const originalChangeStreamEnabled = process.env.CHANGE_STREAM_ENABLED;
+
+  afterAll(() => {
+    process.env.CHANGE_STREAM_ENABLED = originalChangeStreamEnabled;
+    process.env.DELETION_AUDIT_LOGS_TTL_SECONDS = originalDeletionAuditLogsDeleteAfterSeconds;
+  });
+
+  describe('when change stream enabled', () => {
```

Comment:
fwiw id have been happy for this `change stream enabled` describe to just test it calls `deleteDocumentWithAuditLogs` and returns the expected response (and maybe even if `deleteDocumentWithAuditLogs` it just throws it straight up) -- we could then mock `deleteDocumentWithAuditLogs` and we'd have all the test coverage.

No change needed -- just something that might save you some time in unit tests in future :)